internal/model/rwfolder.go

```diff
@@ -644,7 +644,16 @@ func (p *rwFolder) handleDir(file protocol.FileInfo) {
 			if err != nil || p.ignorePermissions(file) {
 				return err
 			}
-			return os.Chmod(path, mode)
+
+			// Stat the directory so we can check its permissions.
+			info, err := osutil.Lstat(path)
+			if err != nil {
+				return err
```

Comment:
Is this the correct way of handling this `Lstat()` error? (It should be very rare that it fails as we have just created the folder as the Syncthing user!)